### PR TITLE
Rewrite delete-interview to use interview list index instead of internship index, fix other bugs

### LINKED ISSUE #169: [PE-D][Tester A] delete-interview shows stale index range in error message after interview is deleted
Steps to reproduce:
1. add Grab /t Software Engineer
2. add-interview 1 /d 2026-12-01 10:00
3. delete-interview 1
4. delete-interview 0

Expected: Error reflecting that there are 0 interviews e.g., "There are no interviews to delete."

Actual: Error says "Invalid index! Please enter a number between 1 and 1"
which is incorrect — no interviews exist at this point.

-------------
Labels: `severity.Low` `type.FunctionalityBug`

### LINKED ISSUE #177: [PE-D][Tester A] delete-interview and update-date use different index systems causing user confusion
Steps to reproduce:
1. add Grab /t Software Engineer
2. add-interview 1 /d 2026-12-01 10:00
3. add Google /t Data Analyst
4. add-interview 2 /d 2026-11-01 09:00
5. list-interview
   → 1. Google - Data Analyst @ 2026-11-01 09:00
   → 2. Grab - Software Engineer @ 2026-12-01 10:00

6. delete-interview 1   → deletes Grab (internship index 1), NOT Google
7. update-date 1        → updates Google (interview list index 1), NOT Grab

Expected: Both commands use the same index system shown in list-interview.
Actual: delete-interview uses internship list index;
        update-date uses interview list index (sorted by date).
A user following list-interview output will delete/update the wrong interview.

-------------
Labels: `severity.High` `type.FunctionalityBug`

### LINKED ISSUE #198: [PE-D][Tester E] Delete-interview using the index for internship not for interview
Although it is clearly stated in the UG, I still think it maybe a litter confusing that the INDEX for delete-interview is the index of the internship whose interview you want to delete but not the index of the interview itself. I feel like using the index of the interview may be more intuitive?

Steps to reproduce:
add Google /t SWE
add Meta /t PM
add-interview 2 /d 2026-06-01 10:00
add-interview 1 /d 2026-07-01 10:00
list-interview — shows index 1 for the Meta interview
delete-interview 1 — user may expect this to delete interview at index 1 per list-interview, but it actually tries to delete the interview of internship at index 1 (Google)
Image

-------------
Labels: `severity.VeryLow` `type.FeatureFlaw`

### PR DESCRIPTION
**Pull request by Yian123330**

### CODE REVIEW

File: src/main/java/seedu/goldencompass/command/DeleteInterviewCommand.java [hunk 1]
```diff
@@ -31,68 +31,70 @@ public void execute() throws GoldenCompassException {
...
         interviewList.getInterviews().remove(interviewToDelete);
 
-        // Also clear the interview reference from the internship
+        // Clear the interview reference from the internship
         internship.deleteInterview();
 
-        // Make sure the message says "Deleted interview", not "Deleted internship"
-        ui.print("Deleted interview for " + internship.getCompanyName()
+        // Print confirmation message
+        ui.print("✓ Deleted interview for " + internship.getCompanyName()
```
Comment (the-RoaringCat) [inline src/main/java/seedu/goldencompass/command/DeleteInterviewCommand.java]: remove the tick

File: src/main/java/seedu/goldencompass/command/DeleteInterviewCommand.java [hunk 2]
```diff
@@ -31,68 +31,70 @@ public void execute() throws GoldenCompassException {
...
-            }
+        if (interviewToDelete == null) {
+            throw new GoldenCompassException("Interview not found.");
         }
 
-        if (interviewToDelete == null) {
-            throw new GoldenCompassException("This internship does not have an interview scheduled.");
+        // Get the associated internship
+        Internship internship = interviewToDelete.getInternship();
+        if (internship == null) {
```
Comment (the-RoaringCat) [inline src/main/java/seedu/goldencompass/command/DeleteInterviewCommand.java:62]: If internship is not found, error is thrown, but interview is not deleted.

File: src/main/java/seedu/goldencompass/command/DeleteInterviewCommand.java [hunk 3]
```diff
@@ -31,68 +31,70 @@ public void execute() throws GoldenCompassException {
...
         }
 
-        int internshipIndex;
+        int interviewIndex;
         try {
-            internshipIndex = Integer.parseInt(params.get(0).trim());
+            interviewIndex = Integer.parseInt(params.get(0).trim());
         } catch (NumberFormatException e) {
             throw new GoldenCompassException("The index must be a number!");
         }
```
Comment (liny666) [inline src/main/java/seedu/goldencompass/command/DeleteInterviewCommand.java:44]: The range check still uses `internshipList.getSize()` instead of `interviewList.size()`. This is the core bug the PR is fixing, but the validation was not updated:
```java
if (interviewIndex < 1 || interviewIndex > interviewList.size()) {
```
As written, if there are 3 interviews but only 1 internship, `delete-interview 2` throws "Invalid index" even though index 2 is valid.

**Review by liny666 — COMMENTED** [review]
Missing test for the new index system — there is no test that verifies `delete-interview 2` deletes the second interview shown by `list-interview` (not the second internship). The existing tests only cover edge cases; a happy-path test with multiple interviews would confirm the core behaviour change is correct.

File: src/main/java/seedu/goldencompass/command/DeleteInterviewCommand.java [hunk 4]
```diff
@@ -31,68 +31,70 @@ public void execute() throws GoldenCompassException {
 
         List<String> params = parser.getParamsOf(parser.getCommand());
 
+        // Check if index is provided
```
Comment (liny666) [inline src/main/java/seedu/goldencompass/command/DeleteInterviewCommand.java:34]: This error message still references the internship: "Please provide the index of the internship to delete interview from!" — should be updated to reference the interview index, consistent with the rest of the rewrite.

File: src/main/java/seedu/goldencompass/command/DeleteInterviewCommand.java [hunk 5]
```diff
@@ -31,68 +31,70 @@ public void execute() throws GoldenCompassException {
...
         } catch (NumberFormatException e) {
             throw new GoldenCompassException("The index must be a number!");
         }
 
-        if (internshipIndex < 1 || internshipIndex > internshipList.getSize()) {
+        // Check if there are any interviews
+        if (interviewList.size() == 0) {
+            throw new GoldenCompassException("There are no interviews to delete.");
+        }
+
```
Comment (liny666) [inline src/main/java/seedu/goldencompass/command/DeleteInterviewCommand.java:50]: This null check is unreachable. `InterviewList.get()` delegates to `interviews.get(index)` which throws `IndexOutOfBoundsException` rather than returning null — so `interviewToDelete` can never be null here. The real guard is the range check above; this is dead code and misleading.

File: src/main/java/seedu/goldencompass/command/DeleteInterviewCommand.java (hunk 2 above)
Comment (Yian123330) [reply src/main/java/seedu/goldencompass/command/DeleteInterviewCommand.java:62]: Good catch. I'll update the code to remove the corrupted interview even if the internship reference is null, so the user can clean up invalid data.

File: src/main/java/seedu/goldencompass/command/DeleteInterviewCommand.java (hunk 3 above)
Comment (Yian123330) [reply src/main/java/seedu/goldencompass/command/DeleteInterviewCommand.java:44]: Fixed. Changed to interviewList.size() instead of internshipList.getSize(). Thanks for catching this.

File: src/main/java/seedu/goldencompass/command/DeleteInterviewCommand.java (hunk 5 above)
Comment (Yian123330) [reply src/main/java/seedu/goldencompass/command/DeleteInterviewCommand.java:50]: Removed the unreachable null check.
